feeds_migrate.module

```diff
@@ -26,22 +26,91 @@ function feeds_migrate_cron() {
   }
 }
 
+/**
+ * Implements hook_migrate_source_info_alter().
+ *
+ * Defines form classes for source plugins.
+ */
+function feeds_migrate_migrate_source_info_alter(array &$plugins) {
+  /* @var \Drupal\feeds_migrate\Plugin\MigrateFormPluginManager $manager */
+  $manager = \Drupal::service('plugin.manager.feeds_migrate.migrate.source_form');
+  $form_plugins = $manager->getDefinitions();
+
+  // Iterate over each source form plugin and attach it to its parent_id plugin.
+  foreach ($form_plugins as $form_plugin) {
+    $parent_plugin_id = $form_plugin['parent_id'] ?? FALSE;
+    $type = $form_plugin['type'] ?? 'configuration';
+    if ($parent_plugin_id && $type && array_key_exists($parent_plugin_id, $plugins)) {
+      $plugins[$parent_plugin_id]['feeds_migrate']['form'][$type] = $form_plugin['class'];
+    }
+  }
+}
+
+/**
+ * Implements hook_migrate_data_fetcher_info_alter().
+ *
+ * Defines form classes for data fetcher plugins.
+ */
+function feeds_migrate_data_fetcher_info_alter(array &$plugins) {
+  /* @var \Drupal\feeds_migrate\Plugin\MigrateFormPluginManager $manager */
+  $manager = \Drupal::service('plugin.manager.feeds_migrate.migrate.data_fetcher_form');
+  $form_plugins = $manager->getDefinitions();
+
+  // Iterate over each data_fetcher form plugin and attach it to its parent plugin.
+  foreach ($form_plugins as $form_plugin) {
+    $parent_plugin_id = $form_plugin['parent_id'] ?? FALSE;
+    $type = $form_plugin['type'] ?? 'configuration';
+    if ($parent_plugin_id && $type && array_key_exists($parent_plugin_id, $plugins)) {
+      $plugins[$parent_plugin_id]['feeds_migrate']['form'][$type] = $form_plugin['class'];
+    }
+  }
+}
+
+/**
+ * Implements hook_migrate_data_parser_info_alter().
+ *
+ * Defines form classes for data parser plugins.
+ */
+function feeds_migrate_data_parser_info_alter(array &$plugins) {
+  /* @var \Drupal\feeds_migrate\Plugin\MigrateFormPluginManager $manager */
```

Comment:
The namespace is currently set to `feeds_migrate`, but you are right that we should have this discussion at some point. Especially when we feel that certain parts of the codebase would ideally end up back in `migrate` or `migrate_tools`